nistats/reporting.py

```diff
@@ -6,16 +6,193 @@
 
 import os
 import warnings
+from string import ascii_lowercase
+
 import numpy as np
+import pandas as pd
 from scipy import stats
 import nilearn.plotting  # overrides the backend on headless servers
+from nilearn.image.resampling import coord_transform
 import matplotlib
 import matplotlib.pyplot as plt
+from scipy import ndimage
 from patsy import DesignInfo
+
 from .design_matrix import check_design_matrix
 matplotlib.rc('xtick', labelsize=20)
 
 
+def _local_max(data, min_distance):
+    """Find all local maxima of the array, separated by at least min_distance.
+    From https://stackoverflow.com/a/22631583/2589328
+
+    Parameters
+    ----------
+    data : array_like
+        3D array of with masked values for cluster.
+
+    min_distance : :obj:`int`
+        Minimum distance between local maxima in ``data``, in terms of
+        voxels (not mm).
+
+    Returns
+    -------
+    xy : :obj:`numpy.ndarray`
+        (n_foci, 3) array of local maxima indices for cluster.
+    """
+    data_max = ndimage.filters.maximum_filter(data, min_distance)
+    maxima = (data == data_max)
+    data_min = ndimage.filters.minimum_filter(data, min_distance)
+    diff = ((data_max - data_min) > 0)
+    maxima[diff == 0] = 0
+
+    labeled, num_objects = ndimage.label(maxima)
+    xy = np.array(ndimage.center_of_mass(data, labeled, range(1, num_objects+1)))
```

Comment:
I'd rather return peak coordinates than center of mass.

Reply:
This is only for neighboring peak voxels with the same value. One thing I noticed with the tests for this function were that some of them used constant blocks of values and the old function would just return the first index, which generally corresponded so a voxel in the corner of the "cluster".